module/project/templates/helper.js

```diff
@@ -957,74 +958,122 @@ module.exports = class ProjectTemplatesHelper {
+                }
+                
+                if( templateId ){
+                    let validateTemplateId = UTILS.isValidMongoId(templateId);
+                    if( validateTemplateId ) {
+                      findQuery["_id"] = templateId;
+                    } else {
+                      findQuery["externalId"] = templateId;
+                    }
                 }
 
-                let templateData = await projectTemplateQueries.templateDocument(findQuery,"all",
-                [
-                    "ratings",
-                    "noOfRatings",
-                    "averageRating",
-                    "parentTemplateId",
-                    "userId",
-                    "createdBy",
-                    "updatedBy",
-                    "createdAt",
-                    "updatedAt",
-                    "__v"
-                ]);
+                //getting template data using templateId
 
+                let templateData = await projectTemplateQueries.templateDocument(findQuery,"all",
+                    [
+                        "ratings",
+                        "noOfRatings",
+                        "averageRating",
+                        "parentTemplateId",
+                        "userId",
+                        "createdBy",
+                        "updatedBy",
+                        "createdAt",
+                        "updatedAt",
+                        "__v"
+                    ]
+                );
+                
                 if ( !templateData.length > 0 ) {
                     throw {
                         status : HTTP_STATUS_CODE.bad_request.status,
-                        message : CONSTANTS.apiResponses.PROJECT_TEMPLATE_NOT_FOUND
-                    }
+                        message :CONSTANTS.apiResponses.PROJECT_TEMPLATE_NOT_FOUND
+                    }    
                 }
 
                 if (templateData[0].tasks && templateData[0].tasks.length > 0) {
                     templateData[0].tasks = 
                     await this.tasksAndSubTasks(templateData[0]._id);
                 }
-
+    
                 let result = await _templateInformation(templateData[0])
-
                 if( !result.success ) {
                     return resolve(result);
                 }
-
-                if( !templateData[0].isReusable ) {
-                    
+    
+                if( !templateData[0].isReusable && userId !== "") {
+                        
                     templateData[0].projectId = "";
-
+    
                     let project = await projectQueries.projectDocument({
                         userId : userId,
                         projectTemplateId : templateData[0]._id
                     },["_id"]);
-
+    
                     if(project && project.length > 0){
                         templateData[0].projectId = project[0]._id;
                     }
                 }
-
+                result.data=solutionsResult;    
```

Comment:
@VISHNUDAS-tunerlabs refer doc and add programInformation key and add program name and _id ,